### PR TITLE
Monitor Tab - New look for empty state + make empty state configurable

### PR DESCRIPTION
**Pull request by nofar9792**
## Which problem is this PR solving?
Resolves https://github.com/jaegertracing/jaeger-ui/issues/907

## Short description of the changes
Add configuration + default configuration.
Use the config in the monitor empty state.
Furthermore, show bar in case the user hasn't sent metrics yet with link to the documentation

before:
image

after:
image

before:
image

after: (with bar)
image

### CODE REVIEW

File: packages/jaeger-ui/src/types/config.tsx
Comment (AlonMiz) [inline packages/jaeger-ui/src/types/config.tsx:79]:
```suggestion
  monitor?: MonitorConfig;
```

File: packages/jaeger-ui/src/components/Monitor/EmptyState/index.tsx
Comment (AlonMiz) [inline packages/jaeger-ui/src/components/Monitor/EmptyState/index.tsx:25]: any? let's be more specific

File: packages/jaeger-ui/src/components/Monitor/EmptyState/index.tsx
Comment (AlonMiz) [inline packages/jaeger-ui/src/components/Monitor/EmptyState/index.tsx:55]:
```suggestion
                {this.config.buttonText}
```

File: packages/jaeger-ui/src/types/config.tsx
Comment (galangel) [inline packages/jaeger-ui/src/types/config.tsx:41]: 👍🏻

**Comment by nofar9792** [comment]
@yuribit @albertteoh WDYT?

**Comment by nofar9792** [comment]
> @yuribit @albertteoh WDYT?

i meant @yurishkuro

**Comment by yurishkuro** [comment]
looks great to me, but I don't have the knowledge to review React code.

File: packages/jaeger-ui/src/constants/default-config.tsx
Comment (albertteoh) [inline packages/jaeger-ui/src/constants/default-config.tsx:102]:
```suggestion
          info: 'Service Performance Monitoring requires a Prometheus-compatible time series database.',
```

**Review by albertteoh — COMMENTED** [review]
Nice work, @nofar9792! I've added a few comments to get yours and others' thoughts on wording and naming. 

Similar to Yuri, I can't comment much on the React code due to lack of knowledge, so trust that @AlonMiz and @galangel had a good look over it.

File: packages/jaeger-ui/src/constants/default-config.tsx
Comment (yurishkuro) [reply packages/jaeger-ui/src/constants/default-config.tsx:93]: +1, I think SPM is a better term that describes the business function, where as ATM describes the implementation.

Comment (yurishkuro) [reply packages/jaeger-ui/src/constants/default-config.tsx:93]: If we do that, let's do a cleanup in the docs too.

File: packages/jaeger-ui/src/constants/default-config.tsx
Comment (nofar9792) [reply packages/jaeger-ui/src/constants/default-config.tsx:100]: changing. BTW we get 404 now
image

File: packages/jaeger-ui/src/constants/default-config.tsx
Comment (nofar9792) [reply packages/jaeger-ui/src/constants/default-config.tsx:93]: changing to Service Performance Monitoring

Comment (albertteoh) [reply packages/jaeger-ui/src/constants/default-config.tsx]: > let's do a cleanup in the docs too.

@yurishkuro, re: cleaning up the docs, besides renaming ATM in docs, were you referring to this? https://github.com/jaegertracing/documentation/issues/541

Or did you have something else in mind?

File: packages/jaeger-ui/src/constants/default-config.tsx
Comment (albertteoh) [reply packages/jaeger-ui/src/constants/default-config.tsx]: > BTW we get 404 now

That's fine, because that page hasn't been released yet, but should be released within 2 weeks, which is when Jaeger UI will also be released.

**Comment by nofar9792** [comment]
@albertteoh @yurishkuro
I fixed everything
I'm waiting for your approval

**Comment by nofar9792** [comment]
i added support in configurable alert type. we will be able to change the alert in the empty state page:
image
image

File: packages/jaeger-ui/src/constants/default-config.tsx
Comment (yurishkuro) [reply packages/jaeger-ui/src/constants/default-config.tsx]: not only that one, Albert already added documentation using the term ATM, so we'd want to change that.

File: packages/jaeger-ui/src/constants/default-config.tsx
Comment (nofar9792) [reply packages/jaeger-ui/src/constants/default-config.tsx]: I missed it.
I'll fix it

Comment (nofar9792) [reply packages/jaeger-ui/src/constants/default-config.tsx]: done

File: packages/jaeger-ui/src/constants/default-config.tsx
Comment (albertteoh) [inline packages/jaeger-ui/src/constants/default-config.tsx:100]: FYI, https://github.com/jaegertracing/documentation/pull/567, so the URL will change to https://www.jaegertracing.io/docs/latest/spm/. 

Alternatively, we can make this change in another PR, as I noticed there are other references to ATM here like `MonitorATMEmptyState`.

Apologies for the back and forth!

**Review by albertteoh — APPROVED** [review]
LGTM, just that we should replace ATM with SPM, but we could do this in another PR.

**Comment by nofar9792** [comment]
@yurishkuro ping..?

**Comment by nofar9792** [comment]
@yurishkuro ?

File: packages/jaeger-ui/src/components/Monitor/ServicesView/index.tsx
Comment (yurishkuro) [inline packages/jaeger-ui/src/components/Monitor/ServicesView/index.tsx:245]:
```suggestion
                No data yet! Please see these
```

File: packages/jaeger-ui/src/components/Monitor/ServicesView/index.tsx
Comment (yurishkuro) [inline packages/jaeger-ui/src/components/Monitor/ServicesView/index.tsx:249]:
```suggestion
                on how to set up your span metrics.
```

File: packages/jaeger-ui/src/constants/default-config.tsx
Comment (yurishkuro) [inline packages/jaeger-ui/src/constants/default-config.tsx:97]:
```suggestion
            'Service Performance Monitoring aggregates tracing data into RED metrics and visualizes them in service and operation level dashboards.',
```

File: packages/jaeger-ui/src/constants/default-config.tsx
Comment (yurishkuro) [inline packages/jaeger-ui/src/constants/default-config.tsx:100]:
```suggestion
            onClick: () => window.open('https://www.jaegertracing.io/docs/latest/spm/'),
```

File: packages/jaeger-ui/src/constants/default-config.tsx
Comment (yurishkuro) [inline packages/jaeger-ui/src/constants/default-config.tsx:107]:
```suggestion
        docsLink: 'https://www.jaegertracing.io/docs/latest/spm/',
```

**Review by yurishkuro — COMMENTED** [review]
- please update the screenshots to reflect later changes
- still need someone with UI/React experience to review this code

**Comment by nofar9792** [comment]
> * please update the screenshots to reflect later changes
> * still need someone with UI/React experience to review this code

@yurishkuro 
Screenshots have been updated
who is going to review my code? @AlonMiz and @galangel already did

**Pull request merged**
